src/main/java/com/gregtechceu/gtceu/api/recipe/lookup/GTRecipeLookup.java

```diff
@@ -373,15 +372,6 @@ protected static void retrieveCachedIngredient(@NotNull List<List<AbstractMapIng
             } else {
                 cache.put(mappedIngredient, new WeakReference<>(mappedIngredient));
             }
-
-            // hardcode a tree specialization for the intersection ingredient
-            if (mappedIngredient instanceof MapIntersectionIngredient intersection) {
-                for (Ingredient inner : intersection.ingredients) {
-                    List<AbstractMapIngredient> converted = ItemRecipeCapability.CAP.convertToMapIngredient(inner);
-                    retrieveCachedIngredient(list, converted, cache);
-                }
-                added = true;
-            }
         }
         if (!added) {
             list.add(ingredients);
```

Comment:
you could also remove the `!added` check from here, and the `added` field itself